src/amo/components/CollectionManager/index.js

```diff
@@ -88,16 +99,30 @@ export class CollectionManagerBase extends React.Component<Props, State> {
       return;
     }
 
-    dispatch(updateCollection({
-      collectionSlug: collection.slug,
-      defaultLocale: collection.defaultLocale,
+    if (!this.state.slug) {
+      errorHandler.addMessage(
+        i18n.gettext('Collection slug cannot be blank')
+      );
+      return;
+    }
+
+    const payload: Object = {
+      collectionSlug: (collection && collection.slug) || undefined,
+      defaultLocale: (collection && collection.defaultLocale) || siteLang,
```

Comment:
I'm assuming here that `siteLang` is what we want to set the initial `defaultLocale` to.